### DISCUSSION TITLE
gpt-4-turbo 带图的问题会"断流"

### DISCUSSION THREAD

**Discussion by zut** [lobehub/lobehub#5164] [General Question]
如果问的问题带图片的:

- 图1: lobe + GPT-4-turbo + 图片:  会断流
- 图1: lobe + GPT-4o + 图片:  正常
- 图2: next + GPT-4-turbo + 图片: 正常 (为了对比,也用其他客户端测试过接口正常)

请问有人遇到过, 或者我应该如何处理呀, 谢谢!

---
图1:
<img width="1240" alt="image" src="https://github.com/user-attachments/assets/d494c9ba-9731-4dde-bd66-ece63520885d" />

图2:
<img width="910" alt="image" src="https://github.com/user-attachments/assets/faa91f4d-20e8-4ee8-80be-d683d3740095" />

**Comment by BrandonStudio** [comment]
开客户端请求，开浏览器控制台，看看响应的event stream

**Reply by zut** [reply]
按你的建议, 这样执行后发现 event stream 的确结束了.

非常奇怪的, 同样的接口, 用其他客户端就不会断掉

<img width="1907" alt="image" src="https://github.com/user-attachments/assets/042a4f97-19b6-4609-99de-c848660ecec1" />
<img width="733" alt="image" src="https://github.com/user-attachments/assets/e664ed12-f507-46e1-9539-726f3f4687e6" />

**Comment by arvinxx** [comment]
我感觉可能是maxToken被限制小了。但按理来说4turbo应该没这个问题，只有4 vision preview才有这个bug

**Reply by zut** [reply]
请看截图, 应该没有被限制到, 内容很少的

<img width="895" alt="image" src="https://github.com/user-attachments/assets/d53d5f60-e21f-45a3-bf73-796cff1c05d4" />

**Reply by BrandonStudio** [reply]
看你的截图，最后一条消息，finish reason是length
说明超出了预设的输出长度，我记得目前默认是2048？还是4096？ @arvinxx

**Reply by arvinxx** [reply]
@zut 你看下其他客户端发出来请求是否会带有 maxToken 的参数？那个是控制单次输出的token长度，和上下文长度是两个东西。

> 说明超出了预设的输出长度，我记得目前默认是2048？还是4096？

目前我们应该没有预设任何的输出长度，应该是直接跟着模型本身的设定来的。

**Reply by arvinxx** [reply]
@zut 据我们之前的经验来看，存在输出截取的只有最早期的 `gpt-4-vision-preview` 这个模型 （ https://github.com/lobehub/lobe-chat/discussions/427#discussioncomment-7540361 ），而 gpt-4-vision-preview 这个模型现在也已经被 OpenAI 废弃，之前的兼容代码也删掉了。至于 gpt-4-turbo 则没有用户反馈过类似的问题。

因此我比较有理由怀疑是你的供应商拿 4-vision-preview 在套 4-turbo ，建议尝试用官方 API key 确认下。如果官方key没有问题就是你的供应商的锅

**Reply by zut** [reply]
@arvinxx 目前看带了max_tokens是正常的, lobe没有带就会中断输出.

<img width="676" alt="image" src="https://github.com/user-attachments/assets/21979af3-ea3a-45bb-84da-7ff4ed34c514" />

**Reply by zut** [reply]
@arvinxx  [ChatGPT-Next-Web](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web) 是会带 max_tokens 参数

**Reply by arvinxx** [reply]
@zut 默认带这个参数并不合理，也不可扩展，因此我们不会考虑默认加的。比如 gpt-4o-mini 最大支持16k的output，默认设成 4000 就很不合理。

你可以手动去设置里拉这个参数到一个较大值，但根本解法应该还是找一个更完备的 api 供应商

**Comment by BrandonStudio** [comment]
> 目前我们应该没有预设任何的输出长度，应该是直接跟着模型本身的设定来的。

@arvinxx 

https://github.com/lobehub/lobe-chat/blob/d5bb78ba9f14a0b7e2486ef33a886d9129fe6cca/src/libs/agent-runtime/bedrock/index.ts#L66

https://github.com/lobehub/lobe-chat/blob/d5bb78ba9f14a0b7e2486ef33a886d9129fe6cca/src/libs/agent-runtime/anthropic/index.ts#L91

https://github.com/lobehub/lobe-chat/blob/d5bb78ba9f14a0b7e2486ef33a886d9129fe6cca/src/libs/agent-runtime/huggingface/index.ts#L25

**Reply by arvinxx** [reply]
要看下不加的情况下输出是否正常，如果正常，这些默认值应该得去掉的

**Reply by BrandonStudio** [reply]
我看主要是 Claude 的部分，但是 3.5 Sonnet 后来不是支持了一个8192的窗口么？没适配吗

**Reply by arvinxx** [reply]
如果代码是这么写的话，估计就没适配吧。
